File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/Tipo_EventoService.java
package com.taekwondo.service;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.taekwondo.model.*;

public interface Tipo_EventoService {
	public abstract List<Tipo_Evento> getTipos_Evento();
	public abstract ResponseEntity<Object> getTipo_Evento(int id);
	public abstract ResponseEntity<Object> createTipo_Evento(Tipo_Evento te);
	public abstract void updateTipo_Evento(int id, Tipo_Evento te);
	public abstract void deleteTipo_Evento(int id);

}

File: proyectoTaeKwonDo/src/app/_services/tipo-evento.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Tipo_Evento} from '../_models/tipo_evento';

@Injectable({
  providedIn: 'root'
})
export class TipoEventoService {

  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getTipos_Evento(){
    return this.http.get(this.API_URI+'/tipo_evento');
  }

  getTipo_Evento(id: number){
    return this.http.get(this.API_URI+'/tipo_evento/'+ id);
  }

  createTipo_Evento(tipoE: Tipo_Evento){
    return this.http.post(this.API_URI + '/tipo_evento', tipoE);
  }

  updateTipo_Evento(tipoE: Tipo_Evento){
    return this.http.put(this.API_URI + '/tipo_evento/' + tipoE.id, tipoE);
  }

  deleteTipo_Evento(id: number){
    return this.http.delete(this.API_URI + '/tipo_evento/'+ id);
  }


}
